boa_engine/src/builtins/intl/date_time_format.rs

```diff
@@ -234,3 +259,1208 @@ pub(crate) fn to_date_time_options(
     // 13. Return options.
     Ok(options)
 }
+
+/// The abstract operation `is_terminal` determines whether `opt` `JsValue` contains a
+/// nonterminal symbol.
+///
+/// More information:
+///  - [Unicode LDML reference][spec]
+///
+/// [spec]: https://www.unicode.org/reports/tr35/#Unicode_locale_identifier
+pub(crate) fn is_terminal(opt: &JsValue, context: &mut Context) -> bool {
+    let opt_str = opt
+        .to_string(context)
+        .unwrap_or_else(|_| JsString::empty())
+        .to_string();
+    if opt_str.is_empty() {
+        return true;
+    }
+
+    // nonterminal = alphanum{3,8} (sep alphanum{3,8})*
+    // Any number of alphanumeric characters between 3 and 8,
+    // separated by dash,
+    // followed by any number of alphanumeric characters between 3 and 8 (repeated)
+
+    // First, replace all underscores (legacy format) with dashes.
+    let opt_str = opt_str.replace('_', "-");
+
+    // Next, split the string by dashes.
+    let options_vec: Vec<&str> = opt_str.split('-').collect();
+
+    // If the vector contains less than 1 element, that cannot be a nonterminal.
+    if options_vec.is_empty() {
+        return true;
+    }
+
+    // Check that each slice is has length between 3 and 8 and all characters are alphanumeric.
+    for option in options_vec {
+        if option.len() < 3 || option.len() > 8 {
+            return true;
+        }
+
+        if option
+            .chars()
+            .any(|character| !character.is_ascii_alphanumeric())
```

Comment:
Wouldn't it be better to do this using the [`all()`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.all) function?

```suggestion
        if !option
            .chars()
            .all(char::is_ascii_alphanumeric)
```

Reply:
Done, however I used lambda since `all(char::is_ascii_alphanumeric)` failed with
```
|   expected signature of `fn(char) -> _`
|   found signature of `for<'r> fn(&'r char) -> _`
```

Is there a better approach for that?

Reply:
I think it's OK this way, the issue is that the `chars()` function returns an iterator with `char` as the item, but the `is_ascii_alphanumeric` expects a reference (`&self`). I think the approach you followed is the best one :)